python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py

```diff
@@ -354,67 +343,58 @@ def get_freshness_policy(cls, dbt_resource_props):
             def get_auto_materialize_policy(cls, dbt_resource_props):
                 return self._node_info_to_auto_materialize_policy_fn(dbt_resource_props)
 
-        (
-            asset_deps,
-            asset_ins,
-            asset_outs,
-            group_names_by_key,
-            freshness_policies_by_key,
-            automation_conditions_by_key,
-            _,
-            fqns_by_output_name,
-            metadata_by_output_name,
-        ) = get_asset_deps(
-            dbt_nodes=dbt_nodes,
-            deps=dbt_dependencies,
-            # TODO: In the future, allow the IO manager to be specified.
-            io_manager_key=None,
+        # generate specs for each executed node
+        dbt_nodes = get_dbt_resource_props_by_dbt_unique_id_from_manifest(manifest_json)
+        specs = build_dbt_asset_specs(
+            manifest=manifest_json,
             dagster_dbt_translator=CustomDagsterDbtTranslator(),
-            manifest=None,
+            select=" ".join(
+                f"fqn:{'.'.join(dbt_nodes[unique_id]['fqn'])}" for unique_id in executed_unique_ids
+            ),
         )
 
         return AssetsDefinitionCacheableData(
             # TODO: In the future, we should allow additional upstream assets to be specified.
-            keys_by_input_name={
-                input_name: asset_key for asset_key, (input_name, _) in asset_ins.items()
-            },
-            keys_by_output_name={
-                output_name: asset_key for asset_key, (output_name, _) in asset_outs.items()
-            },
+            keys_by_output_name={spec.key.to_python_identifier(): spec.key for spec in specs},
```

Comment:
somewhat surprised we don't also have to include checks here, but it doesn't look like we did before either